libevmcore/Assembly.cpp

```diff
@@ -156,6 +166,134 @@ ostream& Assembly::stream(ostream& _out, string const& _prefix, StringMap const&
 	return _out;
 }
 
+Json::Value Assembly::createJsonValue(string _name, int _locationX, int _locationY, string _value, string _jumpType) const
+{
+	Json::Value value;
+	assert(!_name.empty());
+	value["name"] = _name;
+	value["locationX"] = _locationX;
+	value["locationY"] = _locationY;
+	if (!_value.empty())
+		value["value"] = _value;
+	if (!_jumpType.empty())
+		value["jumpType"] = _jumpType;
+	return move(value);
+}
+
+Json::Value Assembly::streamAsmJson(ostream& _out, string const& _prefix, StringMap const& _sourceCodes, bool _inJsonFormat) const
+{
+	Json::Value root;
+
+	string currentArrayName = ".code";
+	std::vector<Json::Value> currentCollection;
+	for (AssemblyItem const& i: m_items)
+	{
+		switch (i.type())
+		{
+		case Operation:
+			currentCollection.push_back(
+						createJsonValue(instructionInfo(i.instruction()).name, i.getLocation().start, i.getLocation().end, i.getJumpTypeAsString()));
+			break;
+		case Push:
+		{
+			std::stringstream hexStr;
+			hexStr << hex << (unsigned)i.data();
```

Comment:
You cannot assume that data() fits into an unsigned. I think u256 should provide a stream interface.